### PR TITLE
Initial benchmark module

### PR DESCRIPTION
**Pull request by rocketraman**
Lots of interesting benchmarks could be added to this e.g. comparing
functional invocation with String concat to the usual idiom, and to
slf4j style patterns.

### CODE REVIEW

**Comment by rocketraman** [comment]
@jvz /review please.

File: log4j-api-kotlin-benchmark/src/main/kotlin/org/apache/logging/log4j/kotlin/benchmark/LoggingBenchmark.kt [hunk 1]
```diff
@@ -0,0 +1,66 @@
+package org.apache.logging.log4j.kotlin.benchmark
```
Comment (jvz) [inline log4j-api-kotlin-benchmark/src/main/kotlin/org/apache/logging/log4j/kotlin/benchmark/LoggingBenchmark.kt:1]: Needs license header.

Comment (rocketraman) [reply log4j-api-kotlin-benchmark/src/main/kotlin/org/apache/logging/log4j/kotlin/benchmark/LoggingBenchmark.kt:17]: Fixed in rebased commits.

**Comment by rocketraman** [comment]
@jvz poke

**Review by jvz — APPROVED** [review]
LGTM. You should have merge permissions here, so feel free to merge this yourself!

**Pull request merged**